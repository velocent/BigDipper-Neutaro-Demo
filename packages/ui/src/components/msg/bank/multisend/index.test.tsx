import renderer from 'react-test-renderer';
import Multisend from '@/components/msg/bank/multisend';
import { MsgMultiSend } from '@/models';
import MockTheme from '@/tests/mocks/MockTheme';

jest.mock('@/components/name', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Name" {...props} />
));

// ==================================
// unit tests
// ==================================
describe('screen: TransactionDetails/MsgMultiSend', () => {
  it('matches snapshot', () => {
    const message = MsgMultiSend.fromJson({
      category: 'bank',
      type: 'MsgMultiSend',
      inputs: [
        {
          address: 'address',
          coins: [
            {
              denom: 'udaric',
              amount: '20000000',
            },
          ],
        },
      ],
      outputs: [
        {
          address: 'output1',
          coins: [
            {
              denom: 'udaric',
              amount: '19000000',
            },
          ],
        },
        {
          address: 'output2',
          coins: [
            {
              denom: 'udaric',
              amount: '1000000',
            },
          ],
        },
      ],
    });

    const component = renderer.create(
      <MockTheme>
        <Multisend message={message} />
      </MockTheme>
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
    expect(
      component.root.findByProps({ i18nKey: 'message_contents:txMultisendContentOne' }).props.values
        .amount
    ).toEqual('20 DARIC');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
