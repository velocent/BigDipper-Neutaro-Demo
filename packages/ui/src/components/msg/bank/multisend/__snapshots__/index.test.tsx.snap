// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TransactionDetails/MsgMultiSend matches snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

.emotion-1 {
  margin-top: 0;
}

<div>
  <p
    className="MuiTypography-root MuiTypography-body1 emotion-0"
  >
    <span
      data-testid="Trans"
      i18nKey="message_contents:txMultisendContentOne"
    >
      <div
        address="address"
        id="Name"
        name="address"
      />
      <b />
      {"amount":"20 DARIC"}
    </span>
  </p>
  <div
    className="emotion-1"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 emotion-0"
    >
      <span
        data-testid="Trans"
        i18nKey="message_contents:txMultisendContentTwo"
      >
        <div
          address="output1"
          id="Name"
          name="output1"
        />
        <b />
        {"amount":"19 DARIC"}
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 emotion-0"
    >
      <span
        data-testid="Trans"
        i18nKey="message_contents:txMultisendContentTwo"
      >
        <div
          address="output2"
          id="Name"
          name="output2"
        />
        <b />
        {"amount":"1 DARIC"}
      </span>
    </p>
  </div>
</div>
`;
